### PR TITLE
switch avatars over to MetaContext

### PR DESCRIPTION
**Pull request by maxtaco**
We're going to be requiring `API#GetDecode` to take a `MetaContext` first argument.  In trying to make that change, I hit resistance here. So pull out the context -> MetaContext change into this PR.

### CODE REVIEW

File: go/avatars/urlcaching.go [hunk 1]
```diff
@@ -53,24 +53,24 @@ func (c *URLCachingSource) isStale(item lru.DiskLRUEntry) bool {
 }
 
 func (c *URLCachingSource) monitorAppState() {
-	c.debug(context.Background(), "monitorAppState: starting up")
+	m := libkb.NewMetaContextBackground(c.G())
+	c.debug(m, "monitorAppState: starting up")
 	state := keybase1.AppState_FOREGROUND
 	for {
 		state = <-c.G().AppState.NextUpdate(&state)
-		ctx := context.Background()
```
Comment (maxtaco) [inline go/avatars/urlcaching.go]: @mmaxim  is there a good reason to make a new context everytime through this loop?  I couldn't figure it out

Comment (mmaxim) [reply go/avatars/urlcaching.go]: No

**Comment by mmaxim** [comment]
Why are these still `libkb.Contextified`? I thought a big reason for this stuff is to get rid of that.

**Comment by maxtaco** [comment]
Happy to change that too, It's a slightly bigger PR (though not much bigger)

**Comment by mmaxim** [comment]
Seems like we should, I feel like that is the prime benefit right?

**Comment by maxtaco** [comment]
OK done. To my mind, the prime benefit is requiring that a `context.Context` (via `libkb.MetaContext`) is passed through to the `API` functions.  Right now, it's kind of there, sometimes.  But that ends up requiring MetaContexts up and down the callstack, which is how I got here.

I'd put de-Contextifying as a secondary benefit, but it doesn't really matter.  Anyways, done!

**Comment by maxtaco** [comment]
Thanks @mmaxim! We'll be psyched about all this when we implement "two simultaneously logged in users"

**Pull request merged**